### ISSUE TITLE
NoSuchMethodError: Class 'SembastStoreRef<int, Map<String, dynamic>>' has no instance method 'add'.

### ISSUE THREAD

**Issue by kwamehubbard** [tekartik/sembast.dart#172]
I get this error message when trying to add/put a new record:
```
flutter: NoSuchMethodError: Class 'SembastStoreRef<int, Map<String, dynamic>>' has no instance method 'add'.
Receiver: Instance of 'SembastStoreRef<int, Map<String, dynamic>>'
Tried calling: add(Instance of 'SembastDatabase', _LinkedHashMap len:1)
```

I've tried several ways to resolve this error including adding
``` 
import 'package:sembast/sembast.dart'
    show
        SembastRecordRefExtension,
        SembastStoreRefExtension
```
Also confirmed that the database is created in the application document folder. I've even go so far as to delete the package from ~/.pub-cache/ and reinstall it Performed a Flutter clean, Pub outdated and Pub upgrade all to no avail.  

In my code snippet I've hard coded a test record thinking I might have an error in my toJson() but both fail so at least I was able to rule out the data. I also noticed that the methods add/put do not appear in Android Studio after typing ```store.```

_As I write this I see that there is an upgrade for Android Studio. I'm on 3.6.3 but will upgrade to 4.0.0 to see if that helps._

My code snippet:
```
store = intMapStoreFactory.store(collectionName);
    try {
      await store.add(db, {'value': 'test'});
    } catch (Exception) {
      print(Exception);
    } finally {
      if (id != null) {
        print('inserted: ' + id.toString());
        feedCount++;
      } else {
        print('document ' + snapshot.documentID + ' not inserted');
      }
    }
```

My Flutter configuration is up to date and Flutter doctor reports no issues:
Flutter 1.17.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 5f21edf8b6 (25 hours ago) • 2020-05-28 12:44:12 -0700
Engine • revision b851c71829
Tools • Dart 2.8.3

**Comment by alextekartik** [comment]
Hi and thanks for the report. Can you try 2 things:
- Check your sdk version in your pubspec.yaml to at least 2.6 or 2.7 (can't remember which version introduces extension) or even 2.8 now since it is in stable

```yaml
environment:
  sdk: ">=2.8.0 <3.0.0"
```

Simple import the whole sembast, using hide to hide conflicting definitions if any

```dart
import 'package:sembast/sembast.dart';
```

**Comment by kwamehubbard** [comment]
Thanks for the suggestions. I changed the sdk version from 2.1.0 to 2.8.0 and imported the whole sembast.dart package. The same error is returned. Also when ahead an upgraded Android Studio to 4.0.0 just to rule it out. 

<img width="609" alt="Screen Shot 2020-05-29 at 7 32 47 PM" src="https://user-images.githubusercontent.com/11198745/83313213-42e25a00-a1e3-11ea-9754-49277ae164cb.png">

**Comment by alextekartik** [comment]
Puzzling to not have the extension suggestion but if you type:
```
await store.add(db, {'value': 'test'});
```

do you get a linter error? a run-time error? using the latest sembast (2.4.4+1)?

**Comment by kwamehubbard** [comment]
No linter error, no run-time error. My app loads and functions (with the exception of adding records to the database). Yes, using the latest sembast. I'm going to try to build a small test harness in just with dart using VS Code and see if I get the same results.

**Comment by kwamehubbard** [comment]
This is a very interesting issue. I fired up VS Code, added the package and copied and pasted the sample code and everything works as expected. The only difference that I saw was that in VS Code the Flutter sdk was at 2.7.0. Everything else was the same. Code assist worked and it did not  throw the error. I changed the sdk to 2.7.0 in AS, did a clean and rebuilt and got the same error.

Back in Android Studio after doing another clean and restarting AS I wanted to see what would happen if I tried intializing ```var store = intMapStoreFactory.store('test'); ``` elsewhere. So I dropped it in ```main.dart``` and this was the result:
<img width="572" alt="Screen Shot 2020-05-30 at 8 15 14 AM" src="https://user-images.githubusercontent.com/11198745/83328005-17488980-a24e-11ea-9814-513b365eb605.png">

For some reason more of the code assist is shown here but not everything. My next step is to fire up a new AS project and see if the results are the same. 

Really appreciate your suggestions but this definitely seems to be an AS issue.

**Comment by kwamehubbard** [comment]
Success! 
Thank you for providing links to those other projects. Ultimately it was by looking at the work of other developers that helped me in resolving this issue. Thank you @baylessj for sharing [autodo](https://github.com/autodo-app/autodo)!

Thanks @alextekartik for the support! 
**Solution**
I had used ```var store = intMapStoreFactory.store('store-name');``` when declaring the store variable. Although this was fine in VS Code, Android Studio could never resolve the ```add``` method. I noticed @bayless used the ```StoreRef``` type instead of ```var```. Once I declared the store using ```StoreRef``` I was able to see all of the methods in code assist and it executed the ```add``` as expected.
